integrations/nuclei_analyzer/app.py

```diff
@@ -0,0 +1,240 @@
+                "success": False,
+                "error": "Failed to run Nuclei",
+                "details": error_msg,
+            }
+
+        output_lines = [
+            line.strip() for line in report.stdout.split("\n") if line.strip()
+        ]
+        parsed_reports = []
+
+        for line in output_lines:
+            try:
+                parsed_reports.append(json.loads(line))
+            except json.JSONDecodeError as e:
+                logger.warning(f"Failed to parse JSON line: {line}, error: {str(e)}")
+                continue
+
+        return True, {
+            "success": True,
+            "result": parsed_reports,
+            "scan_status": "COMPLETED",
+            "statistics": {"total_findings": len(parsed_reports)},
+        }
+
+    except subprocess.TimeoutExpired:
+        return False, {
+            "success": False,
+            "error": "Scan timed out",
+            "scan_status": "TIMEOUT",
+        }
+    except Exception as e:
+        return False, {
+            "success": False,
+            "error": "An unexpected error occurred",
+            "details": str(e),
+            "scan_status": "ERROR",
+        }
+
+
+def async_nuclei_scan(task_id: str, url: str, template_dirs: list = None):
+    try:
+        _, report = run_nuclei_command(url, template_dirs)
+
+        task_store[task_id].update(
+            {
+                "status": "completed",
+                "report": report,
+                "completed_at": datetime.utcnow().isoformat(),
+            }
+        )
+        logger.info(f"Task {task_id} reports stored successfully")
+    except Exception as e:
+        task_store[task_id].update(
+            {
+                "status": "error",
+                "error": str(e),
+                "completed_at": datetime.utcnow().isoformat(),
+            }
+        )
+
+
+@app.route("/run-nuclei", methods=["GET"])
+def get_nuclei_scan_status():
+    task_id = request.args.get("key")
+    if not task_id:
+        return jsonify({"success": False, "error": "Missing task key parameter"}), 400
+
+    if task_id not in task_store:
+        return jsonify({"success": False, "error": "Invalid task key"}), 404
+
+    task_info = task_store[task_id]
+
+    if task_info["status"] == "running":
+        return (
+            jsonify({"status": "running", "started_at": task_info["started_at"]}),
+            200,
+        )
+
+    return (
+        jsonify(
+            {
+                "status": task_info["status"],
+                "report": task_info.get("report"),
+                "error": task_info.get("error"),
+                "started_at": task_info["started_at"],
+                "completed_at": task_info["completed_at"],
+            }
+        ),
+        200,
+    )
+
+
+@app.route("/run-nuclei", methods=["POST"])
+def start_nuclei_scan():
+    try:
+        data = request.get_json()
+
+        if not data or "observable" not in data:
+            return (
+                jsonify(
```

Comment:
try this without jsonify too https://flask.palletsprojects.com/en/stable/quickstart/#about-responses